extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php

```diff
@@ -188,11 +195,40 @@ private function getThemeName( $params, $value ) {
 
 	private function getLayout( $params ) {
 		$layoutName = isset( $params[ 'layout' ] ) ? $params[ 'layout' ] : false;
-		if ( $layoutName && in_array( $layoutName, $this->supportedLayouts ) ) {
+		if ( $this->getParamsValidator()->validateLayout( $layoutName ) ) {
 			//make sure no whitespaces, prevents side effects
 			return self::INFOBOX_LAYOUT_PREFIX . $layoutName;
 		}
 
 		return self::INFOBOX_LAYOUT_PREFIX . self::DEFAULT_LAYOUT_NAME;
 	}
+
+	private function getColor( $colorParam, $params, PPFrame $frame ) {
+		$sourceParam = $colorParam . '-source';
+		$defaultParam = $colorParam . '-default';
+
+		$color = '';
+
+		if ( isset( $params[$sourceParam] ) && !empty( $frame->getArgument( $params[$sourceParam] ) ) ) {
+			$color = trim( $frame->getArgument( $params[$sourceParam] ) );
```

Comment:
It is simple logic and I don't think it will be used in other places so I would leave it as it is